### PR TITLE
Implement the @@isConcatSpreadable well-known Symbol

### PR DESCRIPTION
**Pull request by pmarkee**
Signed-off-by: Peter Marki marpeter@inf.u-szeged.hu

### CODE REVIEW

File: src/runtime/Object.cpp [hunk 1]
```diff
@@ -387,6 +387,24 @@ Object::Object(ExecutionState& state)
...
+    // If Type(O) is not Object, return false.
+    if (!isObject()) {
+        return false;
+    }
+    // Let spreadable be Get(O, @@isConcatSpreadable).
+    ObjectGetResult spreadable = get(state, ObjectPropertyName(state, state.context()->vmInstance()->globalSymbols().isConcatSpreadable));
+    Value val = spreadable.value(state, Value());
+    // If spreadable is not undefined, return ToBoolean(spreadable).
+    if (!val.isUndefined()) {
+        return val.asBoolean();
```
Comment (clover2123) [inline src/runtime/Object.cpp]: In my opinion, `toBoolean()` needs to be called at here instead of `asBoolean()`.

File: src/runtime/Object.h [hunk 2]
```diff
@@ -583,6 +583,8 @@ class Object : public PointerValue {
     }
 #endif
 
+    virtual bool isConcatSpreadable(ExecutionState& state);
```
Comment (clover2123) [inline src/runtime/Object.h]: Should `isConcatSpreadable` be defined as virtual function? If there is no other overriding function, normal function would be more suitable.

**Comment by pmarkee** [comment]
Thanks for the review @clover2123 , I applied your suggestions.

**Comment by yichoi** [comment]
Please enable related tests in https://github.com/pando-tests/js_vendor_tc, if exists.

**Comment by yichoi** [comment]
Please update https://github.com/pando-tests/js_vendor_tc submodule since https://github.com/pando-tests/js_vendor_tc/pull/3 has been landed

**Comment by clover2123** [comment]
Please update the test submodule together with this patch. Thank you.

**Comment by pmarkee** [comment]
@yichoi @clover2123 Done, I updated the submodule.

**Pull request merged**